hashCheck/hashCheck.go

```diff
@@ -0,0 +1,230 @@
+package hashCheck
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/sonalys/fake"
+	"golang.org/x/tools/go/packages"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+const (
+	fileName = "fake.lock.json"
+)
+
+func CompareFileHashes(inputDirs, ignore []string) ([]string, error) {
+	res := make([]string, 0)
+
+	for _, dir := range inputDirs {
+
+		files, err := fake.ListGoFiles(dir, ignore)
+		if err != nil {
+			return nil, fmt.Errorf("getFiles: %w", err)
+		}
+
+		groups := groupByDirectory(files)
+
+		for group, data := range groups {
+			model, err := parseJsonModel(group)
+
+			if err != nil {
+				return nil, fmt.Errorf("parseJsonModel: %w", err)
+			}
+
+			for _, file := range data {
+
+				imports, err := getPackagesGosum(file)
+				if err != nil {
+					return nil, fmt.Errorf("getImportPath: %w", err)
+				}
+
+				goSum, err := hashFiles(imports...)
```

Comment:
`go.sum` already provides a checksum for any given import

Example:
```
github.com/coreos/go-systemd/v22 v22.5.0/go.mod h1:Y58oyj3AT4RCenI/lSvhwexgC+NSVTIJ3seZv2GcEnc=
github.com/godbus/dbus/v5 v5.0.4/go.mod h1:xhWf0FNVPg57R7Z0UbKHbJfkEywrmjJnf7w5xrFpKfA=
github.com/mattn/go-colorable v0.1.13 h1:fFA4WZxdEF4tXPZVKMLwD8oUnCTTo08duU7wxecdEvA=
github.com/mattn/go-colorable v0.1.13/go.mod h1:7S9/ev0klgBDR4GtXTXX8a3vIGJpMovkB8vQcUbaXHg=
github.com/mattn/go-isatty v0.0.16/go.mod h1:kYGgaQfpe5nmfYZH+SKPsOc2e4SrIfOl2e/yFXSvRLM=
github.com/mattn/go-isatty v0.0.19 h1:JITubQf0MOLdlGRuRq+jtsDlekdYPia9ZFsB8h/APPA=
github.com/mattn/go-isatty v0.0.19/go.mod h1:W+V8PltTTMOvKvAeJH7IuucS94S2C6jfK/D7dTCTo3
```

Wouldn't it be more advantageous for us to use this pre-calculated hash for the package? It would be faster, too.

Reply:
If you are having trouble finding `go.sum`, we could modify the [findGoMod](https://github.com/sonalys/fake/blob/ac13f082e00aedd0e9d1e2f0fa7284ebd2785e7e/utils.go#L37)

Reply:
Sounds good, i ll try that